### ISSUE TITLE
`TypeError` in `no-unused-vars` w/ decorators

### ISSUE THREAD

**Issue by nevir** [babel/babel-eslint#281]
```
function decorate() {}

class Foo {
  @decorate(a => a)
  bar() {}
}
```

This triggers the following error:

```
TypeError: Cannot read property 'type' of undefined
    at collectUnusedVariables (<project>/node_modules/eslint/lib/rules/no-unused-vars.js:182:44)
    at collectUnusedVariables (<project>/node_modules/eslint/lib/rules/no-unused-vars.js:216:13)
    at collectUnusedVariables (<project>/node_modules/eslint/lib/rules/no-unused-vars.js:216:13)
    at collectUnusedVariables (<project>/node_modules/eslint/lib/rules/no-unused-vars.js:216:13)
    at collectUnusedVariables (<project>/node_modules/eslint/lib/rules/no-unused-vars.js:216:13)
    at EventEmitter.Program:exit (<project>/node_modules/eslint/lib/rules/no-unused-vars.js:273:30)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.leaveNode (<project>/node_modules/eslint/lib/util/node-event-generator.js:51:22)
    at CodePathAnalyzer.leaveNode (<project>/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:628:23)
    at CommentEventGenerator.leaveNode (<project>/node_modules/eslint/lib/util/comment-event-generator.js:111:23)
    at Controller.traverser.traverse.leave (<project>/node_modules/eslint/lib/eslint.js:872:36)
```

**Comment by danez** [comment]
see #296

**Comment by danez** [comment]
This is fixed in 6.0.4

**Comment by nevir** [comment]
thanks!
